device/barefoot/x86_64-accton_as9516_32d-r0/newport/buffers_defaults_t0.j2

```diff
@@ -14,12 +14,17 @@
 {%- macro generate_buffer_pool_and_profiles() %}
     "BUFFER_POOL": {
         "ingress_lossless_pool": {
+            {%- if dynamic_mode is not defined %}
```

Comment:
Hi
From this PR I see the buffer pool sizes (`ingress_lossless_pool_size`, `ingress_lossy_pool_size`, and `egress_lossy_pool`) are dynamically calculated and the shared headroom pool size (`ingress_lossless_pool.xoff`) is a fixed number.
But in PR Azure/sonic-swss#2317 and Azure/sonic-swss#2323, I see the buffer pool sizes are fixed and the shared headroom pool size is dynamically calculated.
These result in both the buffer pool sizes and the shared headroom pool size are static. Is this a mistake or by intention?
If you are going to have the buffer pool sizes static and the shared headroom pool size dynamic, I would like to suggest
- always defining `size`s in all pools regardless of whether `dynamic_mode` is defined
- defining `xoff` in `ingress_lossless_pool` only if `dynamic_mode` is defined

Reply:
@stephenxs, thank you for the great suggestions! We decided to push the basic version of the feature so far. But of course it should be ready for further improvements. I like your proposals. I would try to follow your second approach with defining XOFF when dynamic mode is enabled, but I believe it would be worth to verify the first one too. The work will be continued by other engineers on our side.